bootcompiler/src/main/scala/org/mozartoz/bootcompiler/Main.scala

```diff
@@ -235,53 +244,60 @@ object Main {
   }
 
   /** Loads one builtin module definition */
-  private def loadModuleDef(prog: Program, moduleDef: File) = {
-    class CC[T] {
-      def unapply(a: Any): Option[T] = Some(a.asInstanceOf[T])
+  private def loadModuleDef(prog: Program, moduleDef: File): List[(String, Record)] = {
+    case class JsParameter(val kind: String)
+
+    case class JsBuiltin(
+      val name: String,
+      val inlineable: Boolean,
+      val inlineOpCode: Option[Int],
+      val params: List[JsParameter]
+    )
+
+    case class JsModule(
+      val name: String,
+      val builtins: List[JsBuiltin]
+    )
+
+    object ModuleJsonProtocol extends DefaultJsonProtocol {
+      implicit val parameterJsonFormat = jsonFormat1(JsParameter)
+      implicit val builtinJsonFormat = jsonFormat4(JsBuiltin)
+      implicit val moduleJsonFormat = jsonFormat2(JsModule)
```

Comment:
These `implicit val` should have explicit result types. Not having them is about to be deprecated in Scala 2.x, and is illegal in Scala 3.x.